cluster/cluster_test.go

```diff
@@ -19,9 +39,103 @@ func TestJoin(t *testing.T) {
 	require.NoError(t, err)
 	defer c.Close()
 
-	t.Run("test registery contains expected services", func(t *testing.T) {
+	t.Run("test registry contains expected services", func(t *testing.T) {
 		services, err := c.Registry.Services(ctx)
 		require.NoError(t, err)
 		require.NotEmpty(t, services["ping"])
 	})
 }
+
+func (suite *ClusterSuite) TestMemberAdd() {
+	t := suite.T()
+
+	cfg, err := ConfigFromFile("./testdata/ping.yml")
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	c, err := Join(ctx, cfg)
+	require.NoError(t, err)
+	defer c.Close()
+
+	t.Run("test member list contains expected number of servers", func(t *testing.T) {
+		members, err := c.MemberList(ctx)
+		require.NoError(t, err)
+		require.Equal(t, 1, len(members))
+	})
+
+	t.Run("test member add successfully adds new member", func(t *testing.T) {
+		lcfg, err := embed.ConfigFromFile(fmt.Sprintf("./testdata/%s", cfg.EtcdConfigFile))
+		require.NoError(t, err)
+
+		LPUrl, err := url.Parse("http://127.0.0.1:22380")
+		require.NoError(t, err)
+
+		LCUrl, err := url.Parse("http://127.0.0.1:22379")
+		require.NoError(t, err)
+
+		APUrl, err := url.Parse("http://127.0.0.1:22380")
+		require.NoError(t, err)
+
+		ACUrl, err := url.Parse("http://127.0.0.1:22379")
+		require.NoError(t, err)
+
+		mecfg := &EtcdConfig{
+			Name:           "node2",
+			DataDir:        "tmp2",
+			LPUrls:         []url.URL{*LPUrl},
+			LCUrls:         []url.URL{*LCUrl},
+			APUrls:         []url.URL{*APUrl},
+			ACUrls:         []url.URL{*ACUrl},
+			InitialCluster: "node2=http://127.0.0.1:22380",
```

Comment:
I was under the impression that all etcd configs in the cluster would have the same initial cluster in their config. This test shows that this node's initial cluster only contains itself. 

This doesn't make much sense to me because from the perspective of the new node. How would it know who's in the actual initial cluster? I'm also guessing that this is the extra information you needed before starting the new node's etcd.

Reply:
yeah, i did something pretty useless here but the method returns the initialcluster information of the cluster the member's trying to join, then the job of the caller is to append their own information onto the initial cluster which in the case above was the `node2` string

Reply:
i decided to do some of the heavy lifting inside `AddMember`, the caller passes in the name and `AddMember` will return the full `InitialCluster` string